cocos/editor-support/creator/CCScale9Sprite.cpp

```diff
@@ -930,7 +936,61 @@ void Scale9SpriteV2::enableTrimmedContentSize(bool isTrimmed) {
 
 void Scale9SpriteV2::setState(State state) {
     this->_brightState = state;
-    //todo add rendering code
+    auto programCache = cocos2d::GLProgramCache::getInstance();
+    if(!programCache) return;
+    if(Scale9SpriteV2::State::DISTORTION == state) {
+        cocos2d::GLProgram* distortionProgram(nullptr);
+            distortionProgram = programCache->getGLProgram(distortionProgramKey);
+            if(!distortionProgram) {
+                //TODO: add implementation;
+                std::string fShader = "";
+                fShader = fShader
```

Comment:
little advice. for best compatibility on GLES, you need a `precision` qualifier for float.